processor/src/decoder/mod.rs

```diff
@@ -206,21 +206,27 @@ impl Process {
 /// * operation index column is used to keep track of the indexes of the currently executing
 ///   operations within an operation group. Values in this column could be between 0 and 8
 ///   (both inclusive) as there could be at most 9 operations in an operation group.
+///
+/// Also keeps track of operations executed when run in debug mode.
 pub struct Decoder {
     block_stack: BlockStack,
     span_context: Option<SpanContext>,
     trace: DecoderTrace,
+    operations: Vec<Operation>,
+    in_debug_mode: bool,
 }
 
 impl Decoder {
     // CONSTRUCTOR
     // --------------------------------------------------------------------------------------------
     /// Returns an empty instance of [Decoder].
-    pub fn new() -> Self {
+    pub fn new(is_debug: bool) -> Self {
```

Comment:
very minor nit: maybe rename `is_debug` to `in_debug_mode` for consistency?

Reply:
Done